src/main/java/io/confluent/connect/jdbc/sink/dialect/DbDialect.java

```diff
@@ -1,320 +0,0 @@
-/*
- * Copyright 2016 Confluent Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package io.confluent.connect.jdbc.sink.dialect;
-
-import org.apache.kafka.connect.data.Date;
-import org.apache.kafka.connect.data.Decimal;
-import org.apache.kafka.connect.data.Schema;
-import org.apache.kafka.connect.data.Time;
-import org.apache.kafka.connect.data.Timestamp;
-import org.apache.kafka.connect.errors.ConnectException;
-
-import java.nio.ByteBuffer;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-
-import javax.xml.bind.DatatypeConverter;
-
-import io.confluent.connect.jdbc.sink.metadata.SinkRecordField;
-import io.confluent.connect.jdbc.util.DateTimeUtils;
-
-import static io.confluent.connect.jdbc.sink.dialect.StringBuilderUtil.Transform;
-import static io.confluent.connect.jdbc.sink.dialect.StringBuilderUtil.joinToBuilder;
-import static io.confluent.connect.jdbc.sink.dialect.StringBuilderUtil.copiesToBuilder;
-
-public abstract class DbDialect {
```

Comment:
The JDBC sink connector already had a form of dialects. They didn't support much functionality, and still used the static `JdbcUtils` class to do most of the work. And, the sink connector used a static method to determine which dialect to use, which meant it couldn't be customized without subclassing the connector and then duplicating a lot of code. A search of GitHub and Google didn't turn up any subclasses or custom sink dialects.

Although the issue was around adding dialects for the _source connector_, I took this opportunity to refactor the existing dialects and have the new extensible `DatabaseDialect` framework support both sources and sinks. This helped to eliminate some duplicate code, and now a single custom `DatabaseDialect` implementation means that the source and sink connector will work with the DBMS.